source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -161,7 +161,17 @@ void RawHttpClientImpl::check(RequestCallbacks& callbacks,
   ASSERT(callbacks_ == nullptr);
   callbacks_ = &callbacks;
 
-  Http::HeaderMapPtr headers = std::make_unique<Http::HeaderMapImpl>(lengthZeroHeader());
+  Http::HeaderMapPtr headers{};
+  const uint64_t request_length =
+      request.attributes().request().http().body().inline_bytes().length();
+  if (request_length > 0) {
+    const Http::HeaderMap& header_map =
```

Comment:
combine this line with the line below, or make this non-reference and use `std::move` below to avoid copy